sdk/include/reactos/debug.h

```diff
@@ -58,6 +61,78 @@ RtlAssert(
     _In_opt_z_ PCHAR Message
 );
 
+#elif defined(_WIN32)
+/* Win32 User-Mode */
+
+#ifndef DEBUG_BUFSIZE
+#define DEBUG_BUFSIZE 512
+#endif
+
+static inline
+ULONG
+__cdecl
+W32DbgPrint(_In_z_ _Printf_format_string_ PCSTR Format, ...)
+{
+    va_list va;
+    char szBuff[DEBUG_BUFSIZE];
+
+    va_start(va, Format);
+
+#ifdef _STRSAFE_H_INCLUDED_
+    StringCchVPrintfA(szBuff, _countof(szBuff), Format, va);
+#else
+    _vsnprintf(szBuff, _countof(szBuff), Format, va);
+#endif
+
+    OutputDebugStringA(szBuff);
+    va_end(va);
+
+    return ERROR_SUCCESS; /* Return STATUS_SUCCESS, since we are supposed to mimic DbgPrint */
+}
+
+#define DbgPrint(Format, ...) W32DbgPrint(Format, __VA_ARGS__)
+#define DbgPrintEx(ComponentId, Level, Format, ...) W32DbgPrint(Format, __VA_ARGS__)
+
+static inline
+__analysis_noreturn
+VOID
+NTAPI
+W32Assert(
+    _In_ PVOID FailedAssertion,
+    _In_ PVOID FileName,
+    _In_ ULONG LineNumber,
+    _In_opt_z_ PCHAR Message)
+{
+    INT id;
+    char szBuff[DEBUG_BUFSIZE];
+
+#ifdef _STRSAFE_H_INCLUDED_
+    StringCchPrintfA(szBuff, _countof(szBuff),
+#else
+    _snprintf(szBuff, _countof(szBuff),
+#endif
+              "*** Assertion failed: %s%s\n"
+              "***   Source File: %s, line %lu\n\n"
+              "(Press Retry to debug the application)\n",
+              (Message ? Message : ""), (PCSTR)FailedAssertion, (PCSTR)FileName, LineNumber);
+
+    OutputDebugStringA(szBuff);
```

Comment:
This is what I was thinking about regarding "sending this message to dprint in case the app runs under a debugger":
```suggestion
    if (IsDebuggerPresent())
        OutputDebugStringA(szBuff);
```
(inspired from the CRT debug helpers)